src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -193,6 +204,7 @@ bool ClusterTaskManager::PoppedWorkerHandler(
   if (canceled) {
     // Task has been canceled.
     RAY_LOG(DEBUG) << "Task " << task_id << " has been canceled when worker popped";
+    erase_from_running_tasks_fn(task_id, scheduling_class);
```

Comment:
Shouldn't we erase on the cancel call?

Reply:
I thought about it but putting it here gives us a central place (not exactly one place since it's also removed in TaskFinishes) to remove it, instead of removing it everywhere which is easy to miss. Like the task is added to the map when we call PopWorker and it's removed during PoppedWorkerHandler.

Reply:
Hmm I see, but it's weird to leave the task running after it's cancelled. How about we be defensive and remove it in the cancel handler and also here?